classes/item.rb

```diff
@@ -1,21 +1,47 @@
+require 'date'
+
 class Item
-  attr_accessor :genre, :author, :source, :label, :publish_date
-  attr_reader :id, :archived
-
-  def initialize(publish_date, archived: false)
-    @id = rand(1..1000)
-    @publish_date = publish_date
-    @archived = archived
-    @label = nil
-    @genre = nil
-    @author = nil
+  attr_accessor :date, :publish_date
+  attr_reader :id, :archived, :label, :author, :genre
+
+  def initialize(label, author, genre, publish_date)
```

Comment:
Kindly remove :label, :author and :genre variables because you already have the setters in methods 🙂

I'm asking this because of this requirement:
_"Exception: properties for the 1-to-many relationships should NOT be set in the constructor method. Instead, they should have a custom setter method created."_